core/src/array_metadata/array_metadata.cc

```diff
@@ -116,7 +116,16 @@ const URI& ArrayMetadata::array_uri() const {
 const Attribute* ArrayMetadata::attribute(unsigned int id) const {
   if (id < attributes_.size())
     return attributes_[id];
+  return nullptr;
+}
 
+const Attribute* ArrayMetadata::attribute(std::string name) const {
+  for (unsigned int i = 0; i < attribute_num();) {
```

Comment:
`++i` missing in the `for` command? Also I would do:
```
auto attribute_num = this->attribute_num();
for(unsigned int i=0; i<attribute_num; ++i)
...
```

Reply:
good catch I will add a test as well